base/promotion.jl

```diff
@@ -222,9 +222,8 @@ minmax(x::Real, y::Real) = minmax(promote(x, y)...)
 # for the multiplication of two types,
 #   promote_op{R<:MyType,S<:MyType}(::typeof(*), ::Type{R}, ::Type{S}) = MyType{multype(R,S)}
 promote_op(::Any)    = (@_pure_meta; Bottom)
-promote_op(::Any, T) = (@_pure_meta; T)
-promote_op{T}(::Type{T}, ::Any) = (@_pure_meta; T)
```

Comment:
The fallback is not necessarily that slow --- in cases we can infer, there is no significant slowdown at all. However there is a chance we will lose some SIMD optimizations (though I haven't analyzed it in detail), so we should keep using the old approach for +, -, *, etc.

I'm also ok with adding back this definition.